src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1683,6 +1683,27 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    // Method Description:
+    // - When the control gains focus, it needs to tell ConPTY about this.
+    //   Usually, these sequences are reserved for applications that
+    //   specifically request SET_FOCUS_EVENT_MOUSE, ?1004h. ConPTY uses this
+    //   sequence REGARDLESS to communicate if the control was focused or not.
+    // - Even if a client application disables this mode, the Terminal & conpty
+    //   should always request this from the hosting terminal (and just ignore
+    //   internally to ConPTY).
+    // - Full support for this sequence is tracked in GH#11682.
+    // - This is related to work done for GH#2988.
+    void ControlCore::GotFocus()
+    {
+        _connection.WriteInput(L"\x1b[I");

```

Comment:
This will mess up non-conpty connections because they likely didn't _request xterm/x10 focus event tracking_.

I almost think we should use a PTY signal for this until we get focus event support plumbed in...

Reply:
(hang it off ITerminalConnection2)

Reply:
You're right that the signal version isn't great _if we get support for focus events._ In the middle ground, however, it leaves us in trouble. :smile:

Reply:
Okay, so here's some crazy bits: https://github.com/microsoft/terminal/compare/dev/migrie/b/2988-focus-fg-with-owner...dev/migrie/f/11682-vt-focus-events

That could all go in right here if wanted, or stacked as separate PRs. The space is now something like 

```mermaid
graph TD
    id1[Allow windows created by console apps to appear above the Terminal #12799] --> id3([Check if FG window is owner])
    id2[Enable the Terminal to tell ConPTY who the owner is #12526] --> id3
    id3 --> id4([FocusEvents in conhost, terminal])  --> id6
    id5[Propagate show/hide window calls against the ConPTY pseudo window to the Terminal #12515] --> id6{{Terminal v1.14}}
```

Reply:
Okay, so this is done now in the PR stacked two on top of this one, #12900. Might as well just do the whole thing.